supply/spec/uploader_spec.rb

```diff
@@ -453,5 +453,152 @@ def ==(other)
+          include_examples 'updates track with correct status and rollout',
+            rollout: nil,
+            release_status: Supply::ReleaseStatus::DRAFT,
+            expected_status: Supply::ReleaseStatus::DRAFT,
+            expected_user_fraction: nil
+        end
+
+        context 'when release_status is IN_PROGRESS' do
+          include_examples 'raises error for invalid rollout',
+            rollout: nil,
+            release_status: Supply::ReleaseStatus::IN_PROGRESS,
+            expected_error: /You need to provide a rollout value when release_status is set to 'inProgress'/
+        end
+
+        context 'when release_status is COMPLETED' do
+          include_examples 'updates track with correct status and rollout',
+            rollout: nil,
+            release_status: Supply::ReleaseStatus::COMPLETED,
+            expected_status: Supply::ReleaseStatus::COMPLETED,
+            expected_user_fraction: nil
+        end
+      end
+
+      context 'when rollout is 0.5' do
+        context 'when release_status is DRAFT' do
+          include_examples 'updates track with correct status and rollout',
+            rollout: 0.5,
+            release_status: Supply::ReleaseStatus::DRAFT,
+            expected_status: Supply::ReleaseStatus::DRAFT,
+            expected_user_fraction: 0.5
+        end
+
+        context 'when release_status is IN_PROGRESS' do
+          include_examples 'updates track with correct status and rollout',
+            rollout: 0.5,
+            release_status: Supply::ReleaseStatus::IN_PROGRESS,
+            expected_status: Supply::ReleaseStatus::IN_PROGRESS,
+            expected_user_fraction: 0.5
+        end
+
+        context 'when release_status is COMPLETED' do
+          include_examples 'updates track with correct status and rollout',
+            rollout: 0.5,
+            release_status: Supply::ReleaseStatus::COMPLETED,
+            # We want to ensure the implementation forces status of IN_PROGRESS when explicit rollout < 1.0 is provided
+            expected_status: Supply::ReleaseStatus::IN_PROGRESS,
+            expected_user_fraction: 0.5
+        end
+      end
+
+      context 'when rollout is 1.0' do
+        context 'when release_status is DRAFT' do
+          include_examples 'updates track with correct status and rollout',
+            rollout: 1.0,
+            release_status: Supply::ReleaseStatus::DRAFT,
+            expected_status: Supply::ReleaseStatus::DRAFT,
+            expected_user_fraction: 1.0
+        end
+
+        context 'when release_status is IN_PROGRESS' do
+          include_examples 'updates track with correct status and rollout',
+            rollout: 1.0,
+            release_status: Supply::ReleaseStatus::IN_PROGRESS,
+            # We want to ensure the implementation forces status of COMPLETED when explicit rollout = 1.0 is provided
+            expected_status: Supply::ReleaseStatus::COMPLETED,
+            expected_user_fraction: nil
+        end
```

Comment:
This should cover the case / request from @mcfarljw in https://github.com/fastlane/fastlane/issues/28995#issuecomment-2703668199